django_socio_grpc/tests/test_permissions.py

```diff
@@ -0,0 +1,204 @@
+                "django_socio_grpc.tests.test_permissions.FakePermission",
+            ],
+        }
+    )
+    def test_settings(self):
+        # test settings correctly passed to grpc_settings
+        self.assertEqual(grpc_settings.DEFAULT_PERMISSION_CLASSES, [FakePermission])
+
+    @mock.patch("django_socio_grpc.services.Service.get_permissions")
+    def test_check_permissions_ok(self, mock_get_permissions):
+        mock_get_permissions.return_value = [FakePermission()]
+        #   Create a dummyservice for unitary tests
+        dummy_service = DummyService()
+        dummy_service.context = FakeContext()
+        #   Call func
+        with mock.patch(
+            "django_socio_grpc.tests.test_permissions.FakePermission.has_permission"
+        ) as mock_has_permissions:
+            mock_has_permissions.return_value = True
+            dummy_service.check_permissions()
+            mock_has_permissions.assert_called_once_with(dummy_service.context, dummy_service)
+
+    @mock.patch("django_socio_grpc.services.Service.permission_denied")
+    @mock.patch("django_socio_grpc.services.Service.get_permissions")
+    def test_check_permissions_not_ok(self, mock_get_permissions, mock_permission_denied):
+        mock_get_permissions.return_value = [FakePermission()]
+        mock_get_permissions.return_value = [FakePermission()]
+        #   Create a dummyservice for unitary tests
+        dummy_service = DummyService()
+        dummy_service.context = FakeContext()
+        #   Call func
+        with mock.patch(
+            "django_socio_grpc.tests.test_permissions.FakePermission.has_permission"
+        ) as mock_has_permissions:
+            mock_has_permissions.return_value = False
+            dummy_service.check_permissions()
+            mock_has_permissions.assert_called_once_with(dummy_service.context, dummy_service)
+            mock_permission_denied.assert_called_once_with(
+                message="fake message", code="fake code"
+            )
+
+    @mock.patch("django_socio_grpc.services.Service.get_permissions")
+    def test_check_object_permissions_ok(self, mock_get_permissions):
+        mock_get_permissions.return_value = [FakePermission()]
+        #   Create a dummyservice for unitary tests
+        dummy_service = DummyService()
+        dummy_service.context = FakeContext()
+        #   Call func
+        with mock.patch(
+            "django_socio_grpc.tests.test_permissions.FakePermission.has_object_permission"
+        ) as mock_has_object_permissions:
+            mock_has_object_permissions.return_value = True
+            dummy_service.check_object_permissions("fake_obj")
+            mock_has_object_permissions.assert_called_once_with(
+                dummy_service.context, dummy_service, "fake_obj"
+            )
+
+    @mock.patch("django_socio_grpc.services.Service.permission_denied")
+    @mock.patch("django_socio_grpc.services.Service.get_permissions")
+    def test_check_object_permissions_not_ok(
```

Comment:
can you check that FakeContext.abort has been correctly call too ?

Reply:
I already have an unitary test for permission_denied method that assert context.abort is called.
In this method I mock permission_denied to make sure it is called with correct args.